### PR TITLE
feat(22554): Add spreadsheet file uploader to the batch subscriptions

### PR DESCRIPTION
**Pull request by vanch3d**
See https://hivemq.kanbanize.com/ctrl_board/57/cards/22554/details/

The PR implements the first step of the batch subscription flow by adding a file uploader to the stepper.

The uploader supports file selection (through the button) and drag-and-drop (within the marked area). 

The current implementation supports the following file formats:
- Microsoft Excel (both `.xls` and `.xslx` file format) 
- Comma-separated values (`.csv`) 

The uploader extracts the FIRST worksheet of the document successfully uploaded and stores the data for processing by the next step of the flow.

### Before 
screenshot-localhost_3000-2024 05 29-12_20_02

### After
screenshot-localhost_3000-2024 05 29-12_20_39

screenshot-localhost_3000-2024 05 29-12_20_53

screenshot-localhost_3000-2024 05 29-12_21_08

### CODE REVIEW

File: hivemq-edge/src/frontend/package.json [hunk 1]
```diff
@@ -84,6 +86,7 @@
     "reactflow": "^11.11.3",
     "tippy.js": "^6.3.7",
     "uuid": "^9.0.1",
+    "xlsx": "https://cdn.sheetjs.com/xlsx-0.20.2/xlsx-0.20.2.tgz",
```
[the remark this answers is not in the document]
Comment (h2xd) [reply hivemq-edge/src/frontend/package.json:89]: It's most likely untyped right? If you need types there we can look into it together.

File: hivemq-edge/src/frontend/src/components/rjsf/BatchSubscription/components/DataSourceStep.tsx [hunk 2]
```diff
@@ -11,16 +18,84 @@ const getDropZoneBorder = (color: string) => {
...
+      {isDragActive ? (
+        <Text>{t('rjsf.batchUpload.dropZone.dropping')}</Text>
+      ) : loading ? (
+        <Text>{t('rjsf.batchUpload.dropZone.loading')}</Text>
+      ) : (
+        <>
+          <Text>{t('rjsf.batchUpload.dropZone.placeholder')}</Text>
+          <Button onClick={open}>{t('rjsf.batchUpload.dropZone.selectFile')}</Button>
+        </>
+      )}
```
Comment (h2xd) [inline hivemq-edge/src/frontend/src/components/rjsf/BatchSubscription/components/DataSourceStep.tsx:97]: suggestion(non-blocking): A bit hard to parse in my head, WDYT about the following?

```suggestion
      {isDragActive && (
        <Text>{t('rjsf.batchUpload.dropZone.dropping')}</Text>
      )}
      {loading && (
        <Text>{t('rjsf.batchUpload.dropZone.loading')}</Text>
      )}
      {!isDragActive && !loading && (
        <>
          <Text>{t('rjsf.batchUpload.dropZone.placeholder')}</Text>
          <Button onClick={open}>{t('rjsf.batchUpload.dropZone.selectFile')}</Button>
        </>
      )}
```

File: hivemq-edge/src/frontend/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (vanch3d) [reply hivemq-edge/src/frontend/package.json:89]: I felt a bit weird indeed but that's how they recommend installing it.
There is an open-source NPM packaging but it has virtually no taker so not sure I want to bind to it

File: hivemq-edge/src/frontend/src/hooks/useEdgeToast/toast-utils.ts [hunk 3]
```diff
@@ -1,5 +1,8 @@
 import { UseToastOptions } from '@chakra-ui/react'
 
+// TODO[NVL] Should be exported
+export type ToastStatus = 'default' | 'success' | 'error' | 'warning' | 'info' | 'loading'
```
[the remark this answers is not in the document]
Comment (vanch3d) [reply hivemq-edge/src/frontend/src/hooks/useEdgeToast/toast-utils.ts:4]: Wrong type from ChakraUI. Corrected
